File: src/day02.nim
import common

proc solve*(input: string): Answer =
  for line in lines input:
    let a = int(line[0]) - int('A')
    let b = int(line[2]) - int('X')
    let p1 = ((4 - a) mod 3 + b) mod 3 * 3 + b+1
    result.part1 += p1
    let p2 = b * 3 + ((a + 2) mod 3 + b) mod 3+1
    result.part2 += p2

File: aoc2022/day02.py
from .utils import Answer

def solve(input: str):
    p1, p2 = 0, 0
    for line in input.split("\n"):
        a = ord(line[0]) - ord('A')
        b = ord(line[2]) - ord('X')
        p1 += ((4 - a) % 3 + b) % 3 * 3 + b+1
        p2 += b * 3 + ((a + 2) % 3 + b) % 3+1

    return Answer(p1, p2)
